src/registry.rs

```diff
@@ -179,9 +182,36 @@ impl RegistryCore {
             });
         }
 
-        // Write out MetricFamilies sorted by their name.
-        let kvs = Vec::from_iter(mf_by_name.into_iter());
-        kvs.into_iter().map(|(_, m)| m).collect()
+        // Write out MetricFamilies sorted by their name (with optional registry values).
+        mf_by_name
+            .into_iter()
+            .map(|(_, mut m)| {
+                if let Some(ref namespace) = self.prefix {
+                    let prefixed = format!("{}_{}", namespace, m.get_name());
+                    m.set_name(prefixed);
+                }
+                if let Some(ref hmap) = self.labels {
+                    let pairs: Vec<proto::LabelPair> = hmap
+                        .iter()
+                        .map(|(k, v)| {
+                            let mut label = proto::LabelPair::new();
+                            label.set_name(k.to_string());
+                            label.set_value(v.to_string());
+                            label
+                        })
+                        .collect();
+
+                    let metrics = m.mut_metric().as_mut_slice();
+                    for entry in metrics {
+                        let labels = entry.mut_label();
+                        for entry in &pairs {
```

Comment:
Maybe we can use [Vec::append](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append) so that you don't need this for loop?

Reply:
The underlying collection is a [RepeatedField](https://docs.rs/protobuf/2.2.4/protobuf/struct.RepeatedField.html), I didn't see a method to append-in-place. Maybe I missed it? Or maybe are are you suggesting to go through something like the following?
```rust
for metric in m.mut_metric().iter_mut() {
  let mut labels = metric.take_label().into_vec();
  labels.append(&mut pairs);
  metric.set_label(labels.into());
}
```

Reply:
How about calling `into_vec()`?

Reply:
Indeed. I've updated the code to do that.